src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/EvalResultDataItem.cs

```diff
@@ -253,7 +253,8 @@ private static DkmEvaluationResultFlags GetFlags(DkmClrValue value, DkmInspectio
                 }
             }
 
-            if (!value.IsError() && value.HasUnderlyingString(inspectionContext))
+            // check for truncated string before HasUndleryingString so we don't pull the string into memory just to set the flag
+            if (!value.IsError() && ((value.EvalFlags & (DkmEvaluationResultFlags)DkmEvaluationResultFlagsExtensions.TruncatedStringFlag) != 0 || value.HasUnderlyingString(inspectionContext)))
```

Comment:
>((value.EvalFlags & (DkmEvaluationResultFlags)DkmEvaluationResultFlagsExtensions.TruncatedStringFlag) != 0 [](http://example.com/codeflow?start=36&length=106)

I think instead you need a `bool` argument to `GetUnderlyingStringImpl` to indicate if you want large strings.

Reply:
Wouldn't `HasUnderlyingString` also check for `TruncatedStringFlag` when allocating the string? 
(...does `GetUnderlyingStringImpl` call into GetValueString?)